_posts/2019-09-23-exposing-http-bridge.md

```diff
@@ -0,0 +1,270 @@
+Using the bridge, it is possible to interact with the Apache Kafka cluster through the HTTP/1.1 protocol instead of the Kafka native one.
+We already covered how it is really simple to deploy the bridge through the new `KafkaBridge` custom resource and via the Strimzi Cluster Operator.
+We also covered all the REST endpoints that the bridge provides as an API for consumers and producers.
+What we are going to describe in this blog post is how it is possible to expose the bridge outside of the Kubernetes (or OpenShift) cluster, where it is running alongside the Kafka cluster.
+We will also show real examples using the `curl` command.
+
+<!--more-->
+
+Before starting with the bridge, the main prerequisite is having a Kubernetes cluster and an Apache Kafka cluster already deployed on top of it using the Strimzi Cluster Operator; to do so you can find information in the official documentation.
+
+# Deploy the HTTP bridge
+
+Deploying the bridge on Kubernetes is really easy using the new `KafkaBridge` custom resource provided by the Strimzi Cluster Operator.
+For the purpose of this post, you can just use the resource provided as an example in Strimzi.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 1
+  bootstrapServers: my-cluster-kafka-bootstrap:9092
+  http:
+    port: 8080
+```
+
+This declaration assumes that the Kafka cluster deployed by the Strimzi Cluster Operator is named `my-cluster` otherwise you have to change the `bootstrapServers` accordingly.
+All the configuration parameters for the consumer and producer part of the bridge are the default ones defined by the Apache Kafka documentation.
+
+Just run the following command in order to apply this custom resource allowing the Strimzi Cluster Operator to deploy the HTTP bridge for you.
+
+```shell
+kubectl apply -f examples/kafka-bridge/kafka-bridge.yaml
+```
+
+# Create a Kubernetes Ingress
+
+An `Ingress` is a Kubernetes resource for allowing external access via HTTP/HTTPS to internal services, like the HTTP bridge.
+Even if you can create an `Ingress` resource, it is possible that it could not work out of the box.
+In order to have an Ingress working properly, an Ingress controller is needed to be running in the cluster.
+There are many different Ingress controllers and Kubernetes already provides two of them:
+
+* [NGINX controller](https://github.com/kubernetes/ingress-nginx)
+* [GCE controller](https://github.com/kubernetes/ingress-gce) for the Google Cloud Platform
+
+> If your cluster is running using Minikube, an NGINX ingress controller is provided as an addon that you can enable with the command `minikube addons enable ingress` before starting the cluster.
+
+Basically, the external traffic reaches the controller which routes the traffic itself based on the rules specified by the `Ingress` resource.
```

Comment:
Isn't the controller more like an operator for the Ingress resources with the Nginx which does the routing running in separate pod(s)?